processor/transformprocessor/internal/metrics/func_agregate_label_metrics.go

```diff
@@ -0,0 +1,409 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type aggregateLabelArguments struct {
+	Type     common.AggregationType
+	LabelSet map[string]bool
+}
+
+type aggGroups struct {
+	gauge        map[string]pmetric.NumberDataPointSlice
+	sum          map[string]pmetric.NumberDataPointSlice
+	histogram    map[string]pmetric.HistogramDataPointSlice
+	expHistogram map[string]pmetric.ExponentialHistogramDataPointSlice
+}
+
+func newAggregateLabelFactory() ottl.Factory[ottlmetric.TransformContext] {
+	return ottl.NewFactory("aggregate_label", &aggregateLabelArguments{}, createAggregateLabelFunction)
```

Comment:
Could you add a test for this function to `internal/metrics/processor_test.go`?

Reply:
Correct me if I am wrong, but for completing the issue, I would need to register the `AggregationType` datatype [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/functions.go#L406) to be able to add this datatype as the first argument for the `aggregation_on_label()`. Afterwards the changes in `pkg/ottl` should get merged, released and firstly then I can use it in the current PR. 

Do we want to go through this process or should we just use `string` for the first argument of `aggregation_on_label()`?

Thanks!

Reply:
One additional question here, what datatype would you suggest for the second parameter for the function? Currently it's `map[string]bool`, but I guess we can simplify it. I do not have much experience regarding this, so I would use a tip for what to use [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33334/files#diff-c1998083baf9788a3013cd95c1711f1d9e48d10ce32b6eea56f4ccca4e9593dfR23).

Thank you!

Reply:
>Do we want to go through this process or should we just use string for the first argument of aggregation_on_label() which we afterwards convert to AggregationType and validate?

That sounds like the best solution. Realistically we may not need to convert it to `AggregationType` for the sake of just this function.

>One additional question here, what datatype would you suggest for the second parameter for the function? 

I think `[]string` would be the best type here.

Reply:
thank you! adapted